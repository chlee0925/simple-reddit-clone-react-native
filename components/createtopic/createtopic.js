import React from 'react';
import { View, Text, TextInput, Button, StyleSheet, Keyboard } from 'react-native';

/**
 * Screen where user can create new topic
 */
export class CreateTopicScreen extends React.Component {
    static navigationOptions = {
        title: 'Create a new topic',
    };

    constructor(props) {
        super(props);
        this.state = {text: ''};
    }

    render() {
        const { params } = this.props.navigation.state;
        return (
            <View>
                <Text style={style.header}>Write your own topic!</Text>
                <TextInput style={style.textInput}
                    placeholder='Type your new topic'
                    onChangeText={(text) => this.setState({text})}
                />
                <Button title='Create Topic' onPress={() => {
                    Keyboard.dismiss();
                    params.createTopic(this.state.text);
                    this.props.navigation.goBack();
                }} />
            </View>
        )
    }
}

const style = StyleSheet.create({
    header: {
        fontSize: 18,
        fontWeight: 'bold',
        margin: 20,
        textAlign: 'center'
    },
    textInput: {
        borderWidth: 1,
        borderColor: 'gray',
        height: 100,
        margin: 10
    }
});
