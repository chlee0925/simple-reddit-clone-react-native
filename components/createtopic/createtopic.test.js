import React from 'react';
import { CreateTopicScreen } from './createtopic.js';

import renderer from 'react-test-renderer';

it('renders CreateTopicScreen Component', () => {
    const mockNavigation = {
        state: {
            params: {
                createTopic: jest.fn()
            }
        },
        goBack: jest.fn()
    }
    const screen = renderer.create(<CreateTopicScreen navigation={mockNavigation} />).toJSON();
    expect(screen).toMatchSnapshot();
});