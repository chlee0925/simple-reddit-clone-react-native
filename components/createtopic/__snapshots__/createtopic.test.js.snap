// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CreateTopicScreen Component 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    Create a new topic!
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Type your new topic"
    style={
      Object {
        "borderWidth": 1,
        "height": 50,
        "margin": 10,
      }
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Create Topic
      </Text>
    </View>
  </View>
</View>
`;
